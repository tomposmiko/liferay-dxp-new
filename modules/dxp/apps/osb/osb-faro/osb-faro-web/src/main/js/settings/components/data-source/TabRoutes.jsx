import BundleRouter from 'route-middleware/BundleRouter';
import omitDefinedProps from 'shared/util/omitDefinedProps';
import React from 'react';
import {PropTypes} from 'prop-types';
import {Switch} from 'react-router-dom';

export default class TabRoutes extends React.Component {
	static propTypes = {
		routes: PropTypes.arrayOf(
			PropTypes.shape({
				component: PropTypes.func,
				path: PropTypes.string
			})
		)
	};

	render() {
		const {routes, ...otherProps} = this.props;

		return (
			<Switch>
				{routes.map(({component: Component, path}) => (
					<BundleRouter
						componentProps={omitDefinedProps(
							otherProps,
							TabRoutes.propTypes
						)}
						data={Component}
						exact
						key={path}
						path={path}
					/>
				))}
			</Switch>
		);
	}
}
