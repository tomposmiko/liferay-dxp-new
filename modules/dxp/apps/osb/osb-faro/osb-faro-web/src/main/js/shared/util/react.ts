import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
