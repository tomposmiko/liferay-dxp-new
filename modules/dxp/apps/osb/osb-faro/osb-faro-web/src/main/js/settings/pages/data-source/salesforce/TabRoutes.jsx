import Authorization from './Authorization';
import ConfigurationStatus from './ConfigurationStatus';
import React from 'react';
import TabRoutes from 'settings/components/data-source/TabRoutes';
import {Routes} from 'shared/util/router';
import {withCurrentUser} from 'shared/hoc';

const TAB_ROUTES = [
	{
		component: withCurrentUser(Authorization),
		path: Routes.SETTINGS_DATA_SOURCE
	},
	{
		component: ConfigurationStatus,
		path: Routes.SETTINGS_SALESFORCE_CONFIGURATION_STATUS
	}
];

export default props => <TabRoutes {...props} routes={TAB_ROUTES} />;
