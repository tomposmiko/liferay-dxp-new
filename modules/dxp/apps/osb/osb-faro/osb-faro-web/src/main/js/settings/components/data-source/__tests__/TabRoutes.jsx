import React from 'react';
import TabRoutes from '../TabRoutes';
import {render} from '@testing-library/react';
import {StaticRouter} from 'react-router';

jest.unmock('react-dom');

describe('TabRoutes', () => {
	it('should render', () => {
		const Component = () => <div>{'Foo Bar'}</div>;

		const {container, getByText} = render(
			<StaticRouter location='foo/path'>
				<TabRoutes
					routes={[{component: Component, path: 'foo/path'}]}
				/>
			</StaticRouter>
		);

		expect(getByText('Foo Bar')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});
