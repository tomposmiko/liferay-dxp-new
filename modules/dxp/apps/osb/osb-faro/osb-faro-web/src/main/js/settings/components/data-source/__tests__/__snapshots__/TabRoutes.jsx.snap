// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabRoutes should render 1`] = `
<div>
  <div>
    Foo Bar
  </div>
</div>
`;
